app/models/category.py

```diff
@@ -1,102 +1,122 @@
-from sqlalchemy import (
-    String,
-    Boolean,
-    Integer,
-    Text,
-    Date,
-    DECIMAL,
-    ForeignKey,
-    Enum,
-)
+from sqlalchemy import String, Boolean, Text, Date, DECIMAL, ForeignKey
 from sqlalchemy.orm import Mapped, mapped_column, relationship
 from uuid import UUID
-from typing import List
+from typing import List, Optional
 from .base import Base
-from .user import User
 from app.enum.category import CategoryNameEnum, CategoryColorEnum
 
 
-# CategoryProgress 테이블 (한 카테고리에 여러 목표, 노트가 연결됨)
-class CategoryProgress(Base):
-    __tablename__ = "category_progresses"
+# ✅ Category 테이블: 사용자의 카테고리를 정의 (코딩/영어/운동 등)
+class Category(Base):
+    __tablename__ = "categories"
 
-    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), nullable=False)
-    category_name: Mapped[CategoryNameEnum] = mapped_column(
-        Enum(CategoryNameEnum, name="category_name_enums"), nullable=False
-    )
-
-    total_goal: Mapped[int] = mapped_column(Integer, nullable=False)
-    completed_goal: Mapped[int] = mapped_column(Integer, nullable=False)
-    progress_rate: Mapped[DECIMAL] = mapped_column(DECIMAL(5, 2), default=0.00)
-
-    # CategoryProgress → User (N : 1 관계)
-    user: Mapped["User"] = relationship("User", back_populates="category_progresses")
+    category_name: Mapped[CategoryNameEnum] = mapped_column(nullable=False)
+    category_color: Mapped[CategoryColorEnum] = mapped_column(nullable=False)
 
-    # CategoryProgress → ComprehensiveEvaluation, RecommendedChallenge, Goal, MonthlyAchievement (1 : N 관계)
-    comprehensive_evaluations: Mapped[
-        List["app.models.evaluation.ComprehensiveEvaluation"]
-    ] = relationship(
-        "app.models.evaluation.ComprehensiveEvaluation",
-        back_populates="category_progress",
+    # 관계: Category 1 : 1 Progress
+    progress: Mapped["app.models.category.Progress"] = relationship(
+        "app.models.category.Progress",
+        back_populates="category",
         cascade="all, delete",
     )
+
+    # 관계: Category 1 : N Todo, Memo, Storage, RecommendedChallenge
```

Comment:
chats 테이블도 관계가 있는 것 같은데 획인해주세용
```suggestion
    # 관계: Category 1 : N Todo, Memo, Storage, Chat, RecommendedChallenge
```

Reply:
@GideokKim 
chat 테이블의 경우 storage 하위에 있기 때문에 storage와 관계를 맺어놓으면 자연스럽게 알 수 있어 또 추가하지 않은거로 알고 있습니다

cc @Ryan-OH @chaeengg @codingorijazz

Reply:
저와 헌찬님이 오전에 미팅해서 상의한 내용이라 채은님 이 방식이 괜찮은지 확인부탁드립니다!